### ISSUE TITLE
docs: build an example of "push to deploy" continuous integration

### ISSUE THREAD

**Issue by eseidel** [shorebirdtech/shorebird#1750]
We've had several customer questions about how best to set up Shorebird.  I think we could offer some docs (or at least a medium article) and a sample repository with a nice setup.  e.g.

What should my release process be?
What should my patch process be?

1. Use GitHub to store the code.
2. Use release branches for making releases.
3. Set up GHA to automatically build patches (and releases?) on push to any release branch or tag?

Questions to answer:
- How can we tell GHA when to make a Release vs a Patch?  Should it always build a Release?  Should it only build a Release if there isn't already one register for Shorebird with that number?

**Comment by eseidel** [comment]
Would be neat to be able to demo "push to deploy" to make a patch.

It might also be interesting to be able to handle "push to deploy" when making a new branch or tag to make a release.

**Comment by mikefaust-jm** [comment]
I recently encountered a similar issue after adding Shorebird to our Flutter app. Since then I have been working on creating a [GitHub Workflow](https://gist.github.com/mikefaust-jm/74df3194ee295cdc281af6f3da4397ac) to automate the release and patch processes for Android and iOS platforms. Currently, it can be triggered by a manual dispatch where you select whether you want a release or a patch. Maybe this helps others to setup their build piplines.

Regarding your question on how to distinguish between making a release and a patch using GitHub Actions, one approach I'm considering is utilizing specific release branch naming conventions and incorporating branch checks within the workflow. For instance:

- Branches named like `release/1.0.5` could trigger a release build, where the build number is derived from `github.run_number`.
- Branches named like `release/1.0.5+6-patch.1` could trigger a patch build for the specific release 1.0.5+6 with the patch number appended.

While this approach isn't perfect, it allows for automation without altering Shorebird's release/patch commands.

However, one challenge I've encountered is regarding the patch numbering, especially when dealing with multiple platforms. Currently, Shorebird increments the patch number after every patch, making it difficult to maintain consistency between platforms. 

One potential solution could be introducing an argument in the patch command to specify a custom patch number for each platform, such as `shorebird patch android --number=1` and `shorebird patch ios --number=1`. 

While searching for a solution I found a similar issue regarding this problem: https://github.com/shorebirdtech/shorebird/issues/1762

Feel free to share any thoughts or suggestions on how we can refine this approach further.
